Objects/abstract.c

```diff
@@ -733,6 +734,41 @@ int PyObject_CopyData(PyObject *dest, PyObject *src)
     return 0;
 }
 
+int
+PyObject_CopyToObject(PyObject *obj, void *buf, Py_ssize_t len,
+                      char Py_UNUSED(fort))
+{
+    Py_buffer view_obj;
+
+    if (!PyObject_CheckBuffer(obj)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "destination must be bytes-like objects");
```

Comment:
The object must only implement the buffer protocol. It does not necessarily need to be a bytes-like object, e.g., the message could be "object supporting the buffer API required"

Reply:
This is to keep in sync with the exception information of `PyObject_CopyData`.

Reply:
But this is wrong. What's required is only a buffer API not bytes-like only.

Reply:
I have made change.